lexorus/flickr-search
# Enhance pod install options

### Background

I was looking through the new cocoapods features (and the old ones) and found a couple of them extremely useful:
1. `preserve_pod_file_structure` - by default the installed pod dependency appears as a folder with files. Sometimes it is useful to persist the file structure for easier navigation through the dependency library.
2. `generate_multiple_pod_projects` - when this is enabled, Xcode project is generated for every dependency, rather than one big `Pods` project for all the dependencies. This is more convenient to work with since Xcode works slower with bigger `.pbxproj` files. As `cocoapods team` suggests, this also should make `pod install` command work faster.
3. `incremental_installation` - this is also performance enhancement. When this is enabled, Xcode project and the target for specific dependency are regenerated only if there are changes from the last `pod install`.

### What has been done?
1. Add option to keep the pod library file structure
2. Enable incremental pod install
3. Enable separation of each pod into its own xcode project
4. Update cocoapods version to 1.9.3

### How to test?
You have to check that this change does not affect in any way the application itself. Try to upgrade/downgrade some of dependencies and check how the pods are updated and projects generated. You should also check that everything works as it is specified above.
